go/vt/sqlparser/ast_funcs.go

```diff
@@ -2686,3 +2686,10 @@ func (node *Update) SetWherePredicate(expr Expr) {
 		Expr: expr,
 	}
 }
+
+func (lock Lock) Merge(newLock Lock) Lock {
```

Comment:
not self-evident what this is doing. comment?

Reply:
renamed the method and added comment.